### ISSUE TITLE
Subtitles menu should close in more cases automatically

### ISSUE THREAD

**Issue by Akxe** [videojs/video.js#5404]
## Description
The whole issue is only apparent if menu was opened via clicking/taping the icon as such it will not close on mouseleave.
The subtitles menu should close on multiple occasion. As of now, they only get closed on an other click on the menu icon or any of it's children. I would suggest that all of these are implemented:

- [x] Click/tap on icon that opened the menu.
- [x] Click/tap on any menu item.
- [ ] Tap/click on video.
  - No other action should be done via closing tap/click
  - Thought I think that click should do the same action as tap in this case, I am not completely settled on it.
- [ ] On user inactive
  - I would even call this current behaviour a bug, as once the control bar disappears, user is still able to click it's content. That can lead to opening subtitle settings, thus confusing user.

Note: This is true for all menus (Menu class)

**Comment by gkatsev** [comment]
Actually, mouseleave does currently close it, but I believe that's actually a bug and not the intended behavior.
Tap/click on the video makes sense. Probably should work when clicking anywhere else if it blurs the menu. This also matches the behavior on youtube.
The fact that the control bar is still clickable during user-inactive is a bug that should be fixed (https://github.com/videojs/video.js/issues/5060). Also, in contrast, youtube doesn't hide the control bar at all when a menu is opened. That may be a reasonable alternative as well.

**Comment by Akxe** [comment]
Yeah, tap on video & disable hiding control bar, when any menu is opened seem like a neat solution. But would live with closing on inactivity too, as if user don't do anything, he probably doesn't care...

**Comment by CoffeyMachine** [comment]
Additionally, it would be ideal to make the click behavior consistent across menus. The playback rate menu, for instance, rotates through the available menu options on click. Why doesn't the subtitles menu do this too?

**Comment by gkatsev** [comment]
I think having the subtitles menu switch between subtitles would be unexpected. In addition, it's an accessibility thing. We want a way to open and pin the menu, especially since the captions settings dialog is inside the menu. If anything were to change, it's likely making the playback rate menu behave like the subtitles menu.

**Comment by CoffeyMachine** [comment]
> If anything were to change, it's likely making the playback rate menu behave like the subtitles menu.

I think this is a good option also.

**Comment by gkatsev** [comment]
We'll consider it but it's unlikely to change in the short term.
